platform/darwin/src/MGLStyle.mm

```diff
@@ -205,12 +206,21 @@ - (void)insertLayer:(MGLStyleLayer *)layer belowLayer:(MGLStyleLayer *)otherLaye
 
 - (void)addSource:(MGLSource *)source
 {
-    self.mapView.mbglMap->addSource(source.mbglSource);
+    // After it is added, the pending source is moved to the mbgl map and is
+    // no longer "pending". The move operation sets it to NULL. However, the
+    // MGLSource's rawSource will still point to the mbgl source and clients
+    // can mutate it via that reference.
+    self.mapView.mbglMap->addSource(source.pendingSource);
```

Comment:
Is it possible to put the `std::move` here, rather than in the `pendingSource` property accessor? That would make it clearer where transfer of ownership is actually occurring.